src/admin/runners/fixDuplicatePractitionerRunner.js

```diff
@@ -0,0 +1,505 @@
+                    query,
+                    { 'meta.lastUpdated': { $gt: this.afterLastUpdatedDate } },
+                    { 'meta.lastUpdated': { $lt: this.beforeLastUpdatedDate } }
+                ]
+            };
+        } else if (this.afterLastUpdatedDate) {
+            query = {
+                $and: [query, { 'meta.lastUpdated': { $gt: this.afterLastUpdatedDate } }]
+            };
+        } else if (this.beforeLastUpdatedDate) {
+            query = {
+                $and: [query, { 'meta.lastUpdated': { $lt: this.beforeLastUpdatedDate } }]
+            };
+        }
+
+        if (this.startFromId) {
+            const startId = isValidMongoObjectId(this.startFromId) ? new ObjectId(this.startFromId)
+                : this.startFromId;
+            query = {
+                       $and: [query, { _id: { $gte: startId } }]
+            };
+        }
+
+        return query;
+    }
+
+    /**
+     * Gets duplicate practitioners with NPI _sourceIds
+     * @param {require('mongodb').collection} collection
+     * @returns {Promise<string[]>}
+     */
+    async getDuplicatePractitionerArrayAsync ({ collection }) {
+        const result = (
+            await collection
+                .aggregate(
+                    [
+                        {
+                            $match: {
+                              _sourceId: {
+                                $regex: '^[0-9]{10}$'
+                              }
+                            }
+                        },
+                        {
+                            $group: {
+                                _id: '$_sourceId',
+                                count: { $count: {} },
+                                meta: { $push: '$meta' },
+                                id: { $push: '$_id' },
+                                uuid: { $push: '$_uuid' },
+                                sourceAssigningAuthority: { $push: '$_sourceAssigningAuthority' }
+                            }
+                        },
+                        {
+                            $match: {
+                                count: {
+                                    $gte: 2
+                                }
+                            }
+                        }
+                    ],
+                    { allowDiskUse: true }
+                )
+                .toArray()
+        );
+
+        return result;
+    }
+
+    /**
+     * Creates map of substitutions for old/new Practitioner values
+     * @param {array} dups
+     * @return {Promise<any>}
+     */
+    async createPractitionerSubstitutions (dups) {
+        // loop thru array and create substitution object fields
+        dups.forEach(dup => {
+            let correctIndex = 0;
+            for (let i = 0; i < dup.sourceAssigningAuthority.length; i++) {
+                if (dup.sourceAssigningAuthority[i] === 'nppes') {
+                    correctIndex = i;
+                }
+            }
```

Comment:
Can we add a condition after this to only process these records if we find a resource with sourceAssigningAuthority nppes, to avoid any other resource whose id matches the pattern but is not related

Reply:
Added code to use that condition